### PR TITLE
Add some examples, including gradient capability for circles/rectangles

### PR DESCRIPTION
**Pull request by mjkalyan**
Note: shapes example 1 is lacking triangles. I figured it would be better to show off that example using only claylib objects.

### CODE REVIEW

File: src/circle.lisp [hunk 1]
```diff
@@ -9,20 +9,29 @@
...
                  :color color
+                 :color2 color2
                  :filled filled))
 
 (defmethod draw-object ((obj circle))
-  (if (filled obj)
-      (claylib/ll:draw-circle-v (c-struct (pos obj))
+  (cond
+    ((and (filled obj) (color2 obj))
+     (claylib/ll:draw-circle-gradient (truncate (x obj))
```
Comment (shelvick) [inline src/circle.lisp:23]: Huh, so apparently we can draw a circle with gradients, or we can draw a circle with fine-tuned float positioning, but not at the same time. This is why I complain about inconsistency in Raylib's API. :-(

**Review by shelvick — CHANGES_REQUESTED** [review]
In core example 28, the ball was moving erratically for me, periodically jumping back and forth across the screen instead of one smooth movement. Did you experience that?

The first two shapes examples looked good; I had some feedback on the third one, primarily the draw loop consing.

File: src/circle.lisp (hunk 1 above)
Comment (mjkalyan) [reply src/circle.lisp:23]: Yes, I noticed your note about that applying to DRAW-CIRCLE-LINES as well. Too bad for now...

**Comment by mjkalyan** [comment]
Besides the intended choppiness when setting a low frame rate, the ball was moving consistently in one direction without jumping erratically for me.

File: examples/shapes/03-colors-palette.lisp [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+                                                    :filled nil
+                                                    :thickness 6))
+                       (draw-object (make-text (nth i color-names)
+                                               (+ (x rect)
+                                                  (width rect)
+                                                  (- (measure-text (nth i color-names) 10))
+                                                  (- 12))
+                                               (+ (y rect) (height rect) (- 20))
+                                               :size 10
+                                               :color (nth i colors)))))))))))
```
[the remark this answers is not in the document]
Comment (mjkalyan) [reply examples/shapes/03-colors-palette.lisp]: Updated to allocate the whole scene beforehand.

File: examples/shapes/03-colors-palette.lisp [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+                                                        :size 10
+                                                        :color +gray+))))))
+      (with-scene scene ()
+        (do-game-loop (:livesupport t
+                       :vars ((color-state (make-array (length colors)
+                                                       :element-type 'bit
+                                                       :initial-element 0))
+                              (mouse-point (make-vector2 0 0))
+                              (faded-rect-color nil)))
+          (setf mouse-point (get-mouse-position mouse-point))
```
[the remark this answers is not in the document]
Comment (mjkalyan) [reply examples/shapes/03-colors-palette.lisp]: Removed the redundancy.

File: examples/core/28-custom-frame-control.lisp [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+                                                        0
+                                                        (floor (/ 1 delta-time)))))
+              (incf time-counter delta-time)))
+
+          ;; Draw
+          ;; ---------------------------------
+          (with-drawing
+            (draw-scene-all scene))
+
+          (claylib/wrap:swap-screen-buffer)
```
[the remark this answers is not in the document]
Comment (mjkalyan) [reply examples/core/28-custom-frame-control.lisp]: I just exported this from `claylib/ll` and it seems to work with no further changes. Is that what you wanted to do?

**Review by shelvick — APPROVED** [review]
Thanks, I'll merge these for now. Core example 28 may be a "me problem," we'll see if anyone else complains. Future improvements: triangle objects, `check-collision-point-rec`.

**Pull request merged**